node/src/transition_frontier/sync/ledger/staged/transition_frontier_sync_ledger_staged_actions.rs

```diff
@@ -194,16 +194,22 @@ impl redux::EnablingCondition<crate::State> for TransitionFrontierSyncLedgerStag
                         TransitionFrontierSyncLedgerStagedState::ReconstructPending { .. }
                     )
                 }),
-            TransitionFrontierSyncLedgerStagedAction::ReconstructSuccess { .. } => state
+            TransitionFrontierSyncLedgerStagedAction::ReconstructSuccess { ledger_hash } => state
                 .transition_frontier
                 .sync
                 .ledger()
                 .and_then(|s| s.staged())
                 .map_or(false, |s| {
+                    // NOTE(tizoc): Assumption here is that the reconstructed staged ledger will always match,
```

Comment:
@binier what do you think of this? best option I can think of is to pass the expected hash to the `staged_ledger_reconstruct` call so that the service can validate it (and trigger an error if it fails)

Reply:
Ignore the above, during staged ledger reconstruction (inside the ledger implementation logic) this is validated already, so it will result in an error, no need to add anything extra.